index.js

```diff
@@ -1077,9 +1077,10 @@ function nested (laterCode, name, key, schema, externalSchema, fullSchema, subKe
         sortedTypes.forEach((type, index) => {
           var tempSchema = Object.assign({}, schema, { type })
           var nestedResult = nested(laterCode, name, key, tempSchema, externalSchema, fullSchema, subKey)
+
           if (type === 'string') {
             code += `
-              ${index === 0 ? 'if' : 'else if'}(typeof obj${accessor} === "${type}" || obj${accessor} instanceof Date || typeof obj${accessor}.toISOString === "function" || obj${accessor} instanceof RegExp)
+              ${index === 0 ? 'if' : 'else if'}(typeof obj${accessor} === "${type}" || obj${accessor} instanceof Date || (obj${accessor} && typeof obj${accessor}.toISOString === "function") || obj${accessor} instanceof RegExp)
```

Comment:
This cover the type:[string] in array issue

```suggestion
              ${index === 0 ? 'if' : 'else if'}(obj${accessor} === null || typeof obj${accessor} === "${type}" || obj${accessor} instanceof Date || typeof obj${accessor}.toISOString === "function" || obj${accessor} instanceof RegExp)
```